target/linux/ipq40xx/files-6.12/arch/arm/boot/dts/qcom/qcom-ipq4029-gx20.dts

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: GPL-2.0-only
+// Device Tree Source for Meraki Go GX20 (Fairyfly)
+
+#include "qcom-ipq4029-wired-qca-common.dtsi"
+
+/ {
+	model = "Meraki Go GX20 Router";
+	compatible = "meraki,gx20";
+	leds {
+		compatible = "gpio-leds";
+
+		power_orange: power {
+			color = <LED_COLOR_ID_AMBER>;
+			function = LED_FUNCTION_POWER;
+			gpios = <&tlmm 49 GPIO_ACTIVE_LOW>;
+			panic-indicator;
+		};
+
+		led-1 {
+			/* WAN left */
+			color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_WAN;
+			function-enumerator = <0>;
+			gpios = <&tlmm 46 GPIO_ACTIVE_LOW>;
+		};
+
+		led-2 {
+			/* WAN right */
+			color = <LED_COLOR_ID_GREEN>;
```

Comment:
I think, that we could also use `LED_FUNCTION_ACTIVITY` and then map the `function-enumerator` to port number. This would cause conflict, though, on devices with WAN and LAN1 port, that can be alleviated by starting from LAN2.

Reply:
In this diff it looks like the mappings for "tx rx" were removed entirely, explaining that, they still need to be mapped, unless we can attach that directly to ports in device tree - but I haven't researched that yet. As I said - I don't insist, as the upstream function definitions aren't perfect either.

Reply:
No worries! If you see similar patterns used in other devices, then I'd prefer consistency over semantic correctness.